python/PiFinder/setup.py

```diff
@@ -163,8 +164,67 @@ def insert_catalog_max_sequence(catalog_name):
     conn.commit()
 
 
+def resolve_object_images():
+    # This is the list of catalogs to search for
+    # objects to match against image names
+    resolution_priority = [
+        "NGC",
+        "IC",
+        "M",
+        "C",
+    ]
+    # load all objects in objects table
+    conn, db_c = objects_db.get_conn_cursor()
+    all_objects = db_c.execute(
+        """
+                SELECT id
+                FROM objects
+            """
+    ).fetchall()
+
+    for obj_record in all_objects:
+        resolved_name = None
+        for catalog_code in resolution_priority:
+            catalog_check = db_c.execute(
+                f"""
+                    SELECT sequence
+                    FROM catalog_objects
+                    WHERE catalog_code = '{catalog_code}'
+                    AND object_id = {obj_record['id']}
+                """
+            ).fetchone()
+            if catalog_check:
+                # Found a match!
+                resolved_name = f"{catalog_code}{catalog_check['sequence']}"
+                break
+
+        if resolved_name == None:
+            # Didn't find a name in the priority list
+            # Pick one from the remainders, but sort
+            # by catalog code so this is deterministic
+            catalog_entry = db_c.execute(
```

Comment:
why not objects_db.get_catalog_object_by_object_id() ?

Reply:
Rewrote this to just continue down the priority order until it finds one which currently yields the same output.